File: proposta/scratch/check_colors.js
const puppeteer = require('puppeteer');
const path = require('path');

(async () => {
    const browser = await puppeteer.launch({
        headless: 'new',
        args: ['--no-sandbox', '--disable-setuid-sandbox']
    });
    const page = await browser.newPage();

    // We load a blank page and draw the image to a canvas to read pixels
    const imgPath = 'file:///' + path.join(__dirname, 'clean_perspective.png').replace(/\\/g, '/');
    console.log('Loading image:', imgPath);

    await page.goto('about:blank');
    
    const colors = await page.evaluate(async (url) => {
        return new Promise((resolve) => {
            const img = new Image();
            img.crossOrigin = 'Anonymous';
            img.onload = () => {
                const canvas = document.createElement('canvas');
                canvas.width = img.width;
                canvas.height = img.height;
                const ctx = canvas.getContext('2d');
                ctx.drawImage(img, 0, 0);
                
                // Sample at X = width / 2 (center) at different Y values
                const x = Math.floor(img.width / 2);
                const samples = {};
                
                // Sample at different vertical positions
                // Y=850 (near bottom of 900px viewport)
                // Y=750 (lower water)
                // Y=650 (mid water)
                // Y=570 (near mountain base)
                // Y=550 (near horizon)
                // Y=400 (sky)
                // Y=100 (zenith sky)
                const yTargets = [100, 400, 550, 570, 650, 750, 850];
                yTargets.forEach(y => {
                    const pixel = ctx.getImageData(x, y, 1, 1).data;
                    samples[y] = `rgb(${pixel[0]}, ${pixel[1]}, ${pixel[2]}) / hex: #${((1 << 24) + (pixel[0] << 16) + (pixel[1] << 8) + pixel[2]).toString(16).slice(1)}`;
                });
                resolve(samples);
            };
            img.src = url;
        });
    }, imgPath);

    console.log('Sampled colors at X = 700 (center of 1400x900 viewport):');
    console.log(JSON.stringify(colors, null, 2));

    await browser.close();
})();

File: proposta/scratch/check_colors.py
import os
from PIL import Image

img_path = os.path.join(os.path.dirname(__file__), 'clean_perspective.png')
print("Opening image:", img_path)

if not os.path.exists(img_path):
    print("Error: Image does not exist!")
    exit(1)

img = Image.open(img_path)
width, height = img.size
print(f"Image dimensions: {width}x{height}")

# Sample at X = width / 2 (center) at different Y values
x = width // 2
y_targets = [100, 400, 550, 570, 650, 750, 850]

print("Sampled colors at X = 700 (center):")
for y in y_targets:
    r, g, b = img.getpixel((x, y))[:3]
    hex_color = f"#{r:02x}{g:02x}{b:02x}"
    print(f"Y={y}: rgb({r}, {g}, {b}) / hex: {hex_color}")
